### PR TITLE
Refs Task #392532 delete user

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## Related Tickets

- [#392532](https://dev.sun-asterisk.com/issues/392532)

## What's this PR do ?

- [x] delete user

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -97,8 +99,17 @@ public function update(Request $request, $id)
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function destroy($id)
+    public function destroy(User $user)
     {
-        //
+        try {
+            if ($user->id == Auth::user()->id) {
+                return redirect()->back()->with('messageFail', 'Delete failed, Cannot delete this user');
```
Comment (tamtt-1031) [inline app/Http/Controllers/UserController.php]: message nên là: ko thể xóa chính mình chứ nhỉ :v

**Comment by phucqd-1305** [comment]
Conflict rồi em

File: app/Http/Controllers/UserController.php (hunk 1 above)
Comment (nguyenthinhhn) [reply app/Http/Controllers/UserController.php]: ok a

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

**Review by tamtt-1031 — APPROVED** [review]

**Comment by ghost** [comment]
xóa user thì các webhooks, bots của user đó có bị xóa không e?

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -111,8 +112,17 @@ public function update(UserUpdateRequest $request, $id)
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function destroy($id)
+    public function destroy(User $user)
     {
-        //
+        try {
+            if ($user->id == Auth::user()->id) {
+                return redirect()->back()->with('messageFail', 'Delete failed, Cannot delete myself');
```
Comment (quanghung97) [inline app/Http/Controllers/UserController.php:119]: cái này có policies rồi. dùng cái đó đi

Comment (nguyenthinhhn) [reply app/Http/Controllers/UserController.php:119]: cái này check middleware admin rồi thì cần gì policies đâu nhỉ.

File: resources/views/modals/delete_user_confirm_modal.blade.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
+                <h4 class="modal-title">Delete Confirmation</h4>
+            </div>
+            <div class="modal-body">
+                <p>Are you sure you want to delete this user?</p>
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-sm btn-danger" id="delete-btn" webhook_id="1">Delete</button>
```
Comment (ghost) [inline resources/views/modals/delete_user_confirm_modal.blade.php]: có cái webhook_id="1" kìa

Comment (nguyenthinhhn) [reply resources/views/modals/delete_user_confirm_modal.blade.php]: em xóa r :)

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

**Review by quanghung97 — APPROVED** [review]
ok

**Review by ghost — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**